### PR TITLE
Fixed sidebar use

### PR DESCRIPTION
**Pull request by igittigitt**
Add some fixes to use this plugin inside a sidebar (regardless which one, the plugin detects for itself).
Also remove some unneccessary code form syntax.php.

### CODE REVIEW

File: syntax.php [hunk 1]
```diff
@@ -13,19 +13,9 @@
...
-    define('DOKU_PLUGIN', DOKU_INC.'lib/plugins/');
-}
-if (!defined('DW_LF')) {
-    define('DW_LF', "\n");
-}
 
-require_once(DOKU_PLUGIN.'syntax.php');
-require_once(DOKU_INC.'inc/parserutils.php');
+// must be run within Dokuwiki
+if (!defined('DOKU_INC')) die();
```
Comment (mprins) [inline syntax.php:18]: if should use brackets

File: syntax.php [hunk 2]
```diff
@@ -96,16 +83,22 @@ function handle($match, $state, $pos, Doku_Handler $handler) {
...
-        if ($mode == 'xhtml') {
+	if ($mode == 'xhtml') {
             $renderer->info['cache'] = false;
 
             @require_once(DOKU_INC.'inc/fulltext.php');
-            $backlinks = ft_backlinks($data[0]);
+	    $page_id = $data[0];
+	    if($page_id == '.') {
+		$page_id = $INFO['id'];
+	    }
```
Comment (mprins) [inline syntax.php:96]: moving this here makes the check specific to xhtml mode and not any others like it did before

File: syntax.php [hunk 3]
```diff
@@ -96,16 +83,22 @@ function handle($match, $state, $pos, Doku_Handler $handler) {
...
+            $backlinks = ft_backlinks($page_id);
 
-            dbglog($backlinks, "backlinks: all backlinks to: $data[0]");
+            dbglog($backlinks, "backlinks: all backlinks to: $page_id");
```
Comment (mprins) [inline syntax.php:99]: why? $page_id is defined and initialized above and makes sure anyone can understand without need to check what is in data and what it means

**Review by mprins — CHANGES_REQUESTED** [review]

File: syntax.php (hunk 1 above)
Comment (igittigitt) [reply syntax.php:18]: Although i agree with you, this line was taken from official syntax-template skeleton. So it should be fine to have it like all other plugins.

File: syntax.php (hunk 2 above)
Comment (igittigitt) [reply syntax.php:96]: Well, this was some kind of hack. My goal was to make the plugin useable inside a sidebar-page.
Therefore the "current page" symbol "." in context of sidebar-page shoud resolve to the displayed wikipage displayed at runtime. Therefore it should not be cached. So i looked for a way to transport this information to the rendering function and used the dot. Maybe we should extend syntax here for usage in sidebar?

File: syntax.php (hunk 3 above)
Comment (igittigitt) [reply syntax.php:99]: Yeah, this could be removed here.

**Comment by mprins** [comment]
Closing this as:
-  non of the requested review notes were addressed, 
- does not provide a test case testing the new code 
- and (even after merging in master) causes test failures of existing test cases

**Pull request closed without merging**